### PR TITLE
[libmad][m1] Crossbuilding not implemented

### PR DESCRIPTION
**Pull request by czoido**
Specify library name and version:  **libmad**

### CODE REVIEW

File: recipes/libmad/all/conanfile.py [hunk 1]
```diff
@@ -23,9 +23,17 @@ def config_options(self):
     def configure(self):
         del self.settings.compiler.libcxx
         del self.settings.compiler.cppstd
-        if self._is_msvc and self.options.shared:
-            raise ConanInvalidConfiguration("libmad does not support shared library for MSVC")
-
+        if self.options.shared:
+            del self.options.fPIC
+            if self._is_msvc:
+               raise ConanInvalidConfiguration("libmad does not support shared library for MSVC")
```
Comment (jgsogo) [inline recipes/libmad/all/conanfile.py:29]: Move this check to `validate`

**Review by jgsogo — COMMENTED** [review]
Why is it not supported? It is because of some error due to cross-building or the error is related to the host configuration?

**Comment by czoido** [comment]
> Why is it not supported? It is because of some error due to cross-building or the error is related to the host configuration?

looks like the host configuration is not valid...

File: recipes/libmad/all/conanfile.py [hunk 2]
```diff
@@ -23,9 +23,17 @@ def config_options(self):
     def configure(self):
         del self.settings.compiler.libcxx
         del self.settings.compiler.cppstd
-        if self._is_msvc and self.options.shared:
-            raise ConanInvalidConfiguration("libmad does not support shared library for MSVC")
-
+ 
+    def validate(self):
+        if self.options.shared:
+            del self.options.fPIC
```
Comment (jgsogo) [inline recipes/libmad/all/conanfile.py:29]: This goes in `configure`, but the check in `validate` 😄

Comment (czoido) [reply recipes/libmad/all/conanfile.py:29]: 🤦‍♂️

**Review by prince-chrismc — APPROVED** [review]

**Review by jgsogo — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Pull request merged**